### PR TITLE
[WIP] Adding ability to load glTF animations

### PR DESCRIPTION
**Pull request by xtanion**
[WIP] This PR is still in progress. 

It adds ability to load animation data from glTF files.

TODO

- Docstrings
- Tests

Check out this [example](https://github.com/xtanion/fury/blob/gltf-anim-merge-kf/docs/tutorials/01_introductory/viz_simple_gltf_animation.py) (merged with PR #626) to see the animation.

### CODE REVIEW

**Comment by m-agour** [comment]
This looks so good! I tested the example and the animation is working. Great job.

**Comment by m-agour** [comment]
Hello @xtanion ,
This looks so good! I tested the example and the animation is working. Great job.

**Comment by skoudoro** [comment]
Thank you @xtanion. Please,  update the description and warn us that the status of this PR is still in work in progress. You still need to do many updates, add some tests, docstring, etc...

File: docs/tutorials/01_introductory/viz_gltf.py [hunk 1]
```diff
@@ -17,8 +17,8 @@
 
 ##############################################################################
 # Retrieving the gltf model.
-fetch_gltf('Duck', 'glTF')
-filename = read_viz_gltf('Duck')
+fetch_gltf('CesiumMan', 'glTF')
```
Comment (skoudoro) [inline docs/tutorials/01_introductory/viz_gltf.py:20]: you have to create a new tutorial. Do not use the same one

**Review by skoudoro — CHANGES_REQUESTED** [review]
Hi @xtanion,

Can you commit your last changes here? 

Also, you need to rebase this PR, and fix the conflicts.

Thank you

**Comment by skoudoro** [comment]
Can you provide some feedback for this @xtanion? thank you

**Comment by xtanion** [comment]
I'm cleaning the PR and documenting it. It shall be done by tomorrow.

**Review by skoudoro — COMMENTED** [review]
Can you add a test for `transform.py`?

Why some interpolators are in `gltf.py`? They should not be there. Can you update?

Thank you

File: fury/gltf.py [hunk 2]
```diff
@@ -73,6 +93,13 @@ def actors(self):
         """
         for i, polydata in enumerate(self.polydatas):
             actor = utils.get_actor_from_polydata(polydata)
+            transform_mat = self.transformations[i]
+            position, rot, scale = transform.trs_from_matrix(transform_mat)
+
+            # We don't need this as we are already applying it in skinnning
```
Comment (skoudoro) [inline fury/gltf.py:99]: comment or/and code to remove if we don't need this

File: fury/gltf.py [hunk 3]
```diff
@@ -106,13 +135,24 @@ def transverse_node(self, nextnode_id, matrix):
...
-
+        if parent is None:
+            parent = [nextnode_id]
+        else:
+            parent.append(nextnode_id)
         matnode = np.identity(4)
+
         if node.matrix is not None:
             matnode = np.array(node.matrix)
+            # print(matnode)
```
Comment (skoudoro) [inline fury/gltf.py:155]: to remove

File: fury/gltf.py [hunk 4]
```diff
@@ -162,7 +206,8 @@ def load_mesh(self, mesh_id, transform_mat):
             attributes = primitive.attributes
 
             vertices = self.get_acc_data(attributes.POSITION)
-            vertices = transform.apply_transfomation(vertices, transform_mat)
+            # vertices = transform.apply_transfomation(vertices, transform_mat)
```
Comment (skoudoro) [inline fury/gltf.py:209]: to remove

File: fury/transform.py [hunk 5]
```diff
@@ -341,3 +341,21 @@ def apply_transfomation(vertices, transformation):
     vertices = vertices[:, :shape[1]]
 
     return vertices
+
+
+def trs_from_matrix(transform):
```
Comment (skoudoro) [inline fury/transform.py:346]: missing docstring. to rename `transform_from_matrix`

**Review by skoudoro — COMMENTED** [review]
Hi @xtanion,

Can you also fix the typo for `apply_transfomation`. it is missing the `r`in transformation

**Comment by skoudoro** [comment]
this PR needs still some work but I will go ahead and merge it. 

Can you add a test for `interpolator.py` in a new PR @xtanion. Thanks

**Pull request merged**